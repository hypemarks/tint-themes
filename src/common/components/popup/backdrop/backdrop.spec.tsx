import { render, screen } from '@testing-library/react';
import React from 'react';
import Backdrop from './backdrop';
import userEvent from '@testing-library/user-event';

describe('Test <Backdrop /> Component', () => {
  it('should render snapshot', () => {
    const wrapper = render(<Backdrop onClick={jest.fn} />);
    expect(wrapper.asFragment()).toMatchSnapshot();
  });

  it('should called onClick callback', () => {
    const onclick = jest.fn();
    render(<Backdrop onClick={onclick} />);
    userEvent.click(screen.getByTestId('backdrop'));
    expect(onclick).toHaveBeenCalledTimes(1);
  });

  it('should change style if disabled flag is not passed', () => {
    const wrapper = render(<Backdrop onClick={jest.fn} lightboxColor={'red'} />);
    const backdrop = wrapper.queryByTestId('backdrop');
    expect(backdrop).toHaveStyle({ background: 'red' });
  });
});
