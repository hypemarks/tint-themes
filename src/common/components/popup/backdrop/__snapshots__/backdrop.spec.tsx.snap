// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <Backdrop /> Component should render snapshot 1`] = `
<DocumentFragment>
  <div
    class="backdrop"
    data-testid="backdrop"
    style="background: rgb(0, 0, 0);"
  />
</DocumentFragment>
`;
